code/modules/mining/mine_turfs.dm

```diff
@@ -1078,3 +1130,139 @@ turf/unsimulated/mineral/ChangeTurf(var/turf/N, var/tell_universe=1, var/force_l
 	if(locate(/obj/structure/lattice) in contents)
 		return BUILD_SUCCESS
 	return BUILD_FAILURE
+
+/turf/simulated/floor/asteroid //For xeno nest specials
+	name = "Asteroid"
+	icon = 'icons/turf/floors.dmi'
+	icon_state = "asteroid"
+	oxygen = 0.01
+	nitrogen = 0.01
+	temperature = TCMB
+	var/dug = 0       //0 = has not yet been dug, 1 = has already been dug
+	var/mined_type = /turf/unsimulated/floor/asteroid
+
+/turf/simulated/floor/asteroid/canBuildCatwalk()
+	return BUILD_FAILURE
+
+/turf/simulated/floor/asteroid/canBuildLattice()
+	if(src.x >= (world.maxx - TRANSITIONEDGE) || src.x <= TRANSITIONEDGE)
```

Comment:
This is not my code, this is old code that I didn't even touch, so I'm not maintaining it